internal/platform/platform.go

```diff
@@ -24,13 +31,55 @@ func (hp *HardwarePlatform) PciDevices() ([]*ghw.PCIDevice, error) {
 	return pciInfo.Devices, nil
 }
 
+func (hp *HardwarePlatform) NetDevs() ([]*ghw.NIC, error) {
+	netInfo, err := ghw.Network()
+	if err != nil {
+		return nil, err
+	}
+	return netInfo.NICs, nil
+}
+
 func (hp *HardwarePlatform) Product() (*ghw.ProductInfo, error) {
 	return ghw.Product()
 }
 
+func (hp *HardwarePlatform) ReadDeviceSerialNumber(pciDevice *ghw.PCIDevice) (string, error) {
+	if pciDevice == nil {
+		return "", fmt.Errorf("nil PCI device provided")
+	}
+
+	devicePath := filepath.Join("/sys/bus/pci/devices", pciDevice.Address, "config")
+
+	file, err := os.Open(devicePath)
+	if err != nil {
+		return "", fmt.Errorf("failed to open config space: %v", err)
+	}
+	defer file.Close()
+
+	// Seek to offset 0x150
+	// Capabilities: [150] Device Serial Number 88-dc-97-ff-ff-44-24-8b
+	const serialOffset = 0x150
+	_, err = file.Seek(serialOffset, 0)
```

Comment:
ghw.PCIDevice didn't have this capability?

Reply:
Unfortunately no... It doesn't read extended capabilities.